### ISSUE TITLE
Comments fail to load correctly

### ISSUE THREAD

**Issue by UmBottesWillen** [TeamNewPipe/NewPipe#10996]
### Checklist

- [X] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.27.0

### Steps to reproduce the bug

1) Long-press on a video of a channel (that has a pinned comment that can be expanded*)
2) Start the video playback in the background
3) (Expand the pinned comment)
4) Minimize the video
5) Long-press on another video of the same channel (that also has a pinned comment that can be expanded*)
6) Start the playback of that video in the background as well
7) Observe that the comments under the new video are still the ones under the old video. The comments of the new video have not been loaded

*) The comment has more than two lines and can be clicked on to be shown in full, not the new 'show answers' feature.

### Expected behavior

When the playback of the new video is started, the comments from that new video are loaded.

### Actual behavior

When the playback of the new video is started, the comments of the old video are still displayed.

### Affected Android/Custom ROM version

Android 13

**Comment by somos61** [comment]
hmm, i have always the comments section of exact video playing, regardless of the player type i choos:, mini, popup or background. v0.27.0

**Comment by somos61** [comment]
> @somos61 Have you tried the exact STR given by OP?

yes

**Comment by somos61** [comment]
found this case interesting because sometimes it would be nice to listen/watch one video and read comments/info of another. Thought it came possible with this release, but no, still not..

**Comment by somos61** [comment]
@UmBottesWillen let me guess one thing, i found that many channels on yt has a copy of the same one comment first in a row each new video next. Check again is that your case, look at the video card name string when starting new background, is it changes or not

**Comment by UmBottesWillen** [comment]
The network does not have any bearing on the bug. I don't have another device to test this bug on. I experienced this bug with videos from the following channel: https://www.youtube.com/channel/UCPVNT9hahuwpVwEpAgO0BtQ
Notably these two videos: 
 - 1) https://www.youtube.com/watch?v=T7AT5wIrmcA
 - 2) https://www.youtube.com/watch?v=U2hOXsM1VMo

**Comment by UmBottesWillen** [comment]
A video demonstrating the issue:

**Comment by somos61** [comment]
yes, confirm an issue with this exact two videos, and first pinned comment expanded.
@UmBottesWillen do you have an error in notifications from NP?

**Comment by UmBottesWillen** [comment]
> @UmBottesWillen Your video link is broken.
> 

The conversion has broken the video, at least on here. It worked when I downloaded it. Can't do any better with the 10Mb upload limit.

> @UmBottesWillen do you have an error in notifications from NP? because i had but i closed it and cant catch it again..

Haven't seen any error, no.

**Comment by somos61** [comment]
Dunno if its a big deal for you, but as a workaround you can always tap on video in channel list while in background play to load its comments and so on. You found rare ones indeed. Tried various chann videos with timecodes in pinned comments - no issues at all. But for this exact chann all tested with timecodes seem issued. And i suspect it is not caused by timecodes itself. Chars/emojis perhaps, who knows, maybe not about the comments at all. Try not to expand the comment and see what happens, for me it is still issued

**Comment by somos61** [comment]
hmm, indeed, named the same..

**Comment by UmBottesWillen** [comment]
I have edited the OP with the updates to the steps to reproduce (5 days ago).

**Comment by UmBottesWillen** [comment]
No, it occurs for everything related to that video. The related videos tab, as well as the description tab and the thumbnail shown aren't updated either. The only thing that updates correctly is the mini-thumbnail when the video is minimized at the bottom of the screen.

**Comment by swever826** [comment]
It started happening to me on the latest nightly version of the app. Only the comments tab seems to show the ones of the previous video instead of the one I am watching.

App version: 0.28.0-33-20251 0070125
OS: Android 16
Device: Pixel 7a

Video: https://streamable.com/cm6jf9 (github didn't let me upload the video so I uploaded in that website)

Edit: I've seen that when I open a video without it playing automatically, it loads the correct comments of the current video, but when I tap on the video and it starts to load and play, the comments tab reloads and shows the comments of the previous video I watched.
